diff -ur apt-cacher-ng-orig/conf/acng.conf.in apt-cacher-ng/conf/acng.conf.in
--- apt-cacher-ng-orig/conf/acng.conf.in	2021-12-07 16:43:56.000000000 +0200
+++ apt-cacher-ng/conf/acng.conf.in	2023-10-10 22:41:57.423129785 +0300
@@ -403,7 +403,7 @@
 # Default: guessed on startup by looking for explicit mention of apt-cacher-ng
 # in /etc/hosts.allow or /etc/hosts.deny files.
 #
-# UseWrap: 0
+UseWrap: 0
 
 # If many machines from the same local network attempt to update index files
 # (apt-get update) at nearly the same time, the known state of these index file
diff -ur apt-cacher-ng-orig/src/acfg.h apt-cacher-ng/src/acfg.h
--- apt-cacher-ng-orig/src/acfg.h	2021-12-07 16:43:56.000000000 +0200
+++ apt-cacher-ng/src/acfg.h	2023-10-10 22:41:57.423129785 +0300
@@ -6,6 +6,7 @@
 #include "actypes.h"
 #include <map>
 #include <bitset>
+#include <sys/time.h>
 
 #define NUM_PBKDF2_ITERATIONS 1
 // 1757961
diff -ur apt-cacher-ng-orig/src/ahttpurl.cc apt-cacher-ng/src/ahttpurl.cc
--- apt-cacher-ng-orig/src/ahttpurl.cc	2021-12-07 16:43:56.000000000 +0200
+++ apt-cacher-ng/src/ahttpurl.cc	2023-10-10 22:43:09.353594226 +0300
@@ -32,7 +32,7 @@
 		bSSL=true;
 #endif
 	}
-	else if(isalnum((uint)url[0]))
+	else if(isalnum((int)url[0]))
 		hStart=0;
 	else if(url[0]=='[')
 	{
diff -ur apt-cacher-ng-orig/src/meta.h apt-cacher-ng/src/meta.h
--- apt-cacher-ng-orig/src/meta.h	2021-12-07 16:43:56.000000000 +0200
+++ apt-cacher-ng/src/meta.h	2023-10-10 22:41:57.423129785 +0300
@@ -20,6 +20,7 @@
 #include <strings.h>
 #include <cstdlib>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "astrop.h"
 
